app/src/main/java/com/example/trainingintive/viewmodels/SplashViewModel.kt

```diff
@@ -1,18 +1,19 @@
 package com.example.trainingintive.viewmodels
 
 import androidx.lifecycle.ViewModel
-import com.example.trainingintive.navigators.Navigator
 import com.example.trainingintive.navigators.SplashNavigator
 import com.example.trainingintive.util.SplashScreenEvent
 import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
 import io.reactivex.rxjava3.core.Single
 import io.reactivex.rxjava3.disposables.Disposable
 import io.reactivex.rxjava3.schedulers.Schedulers
 import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
 
-class SplashViewModel : ViewModel() {
+@Singleton
```

Comment:
ViewModel can't be a singleton because its lifecycle will be tied to an application instead of activity.

Reply:
Right